float64 interval
uint32 encoderR
uint32 encoderL
bool chksum


